src/workerd/api/web-socket.c++

```diff
@@ -757,9 +786,12 @@ size_t countBytesFromMessage(const kj::WebSocket::Message& message) {
 } // namespace
 
 kj::Promise<void> WebSocket::pump(
-    IoContext& context, OutgoingMessagesMap& outgoingMessages, kj::WebSocket& ws, Native& native) {
+    IoContext& context, OutgoingMessagesMap& outgoingMessages, kj::WebSocket& ws, Native& native,
+    kj::Maybe<kj::Promise<void>&> maybeOngoingAutoResponse, bool& isPumping, bool& isClosed,
```

Comment:
non-blocking nit: Is there any way these can be coalesced into a single struct that is passed by reference rather than individual arguments? This pattern always makes me a bit nervous and passing a lot of individual args by reference feels like an anti-pattern to me.

Reply:
Yup, that's the plan. Had a `TODO(cleanup)` for that.